packages/core/integration-tests/test/javascript.js

```diff
@@ -2060,6 +2060,114 @@ describe('javascript', function () {
     assert.deepEqual(await (await run(b)).default, [3, 3]);
   });
 
+  it('does not delete a child bundle, type change or async, of a dynamic import that gets internalized', async () => {
+    let b = await bundle(
+      path.join(__dirname, '/integration/internalize-no-orphan-child/index.js'),
+    );
+
+    assertBundles(b, [
+      {
+        name: 'index.js',
+        assets: [
+          'index.js',
+          'bar.js',
+          'foo.js',
+          'esmodule-helpers.js',
+          'bundle-url.js',
+          'cacheLoader.js',
+          'js-loader.js',
+        ],
+      },
+      {
+        assets: ['bazz.js'], //Async child of internalized foo
+      },
+      {
+        assets: ['a.css'], //Type change child of internalized foo
+      },
+    ]);
+
+    assert.deepEqual(await (await run(b)).default, [3, 3]);
+  });
+
+  // case where the bundle that recieves the newly orphaned type change bundles, also, had a type change bundle, in which case we merge
+  it('does not delete a child bundle, type change or async, of a dynamic import that gets internalized', async () => {
```

Comment:
Nit: should this test description reflect the comment above it? It looks like it's the same as the test above.